### PR TITLE
create nossence bot service

### PR DESCRIPTION
**Pull request by toruta39**
Create a bot service that:

- listens to `#subscribe` messages
- create sub accounts to corresponding user
- send user a welcome message on sub account creation

### CODE REVIEW

File: server/bot/bot.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	}
+}
+
+func (ba *BotApplication) Run(ctx context.Context) error {
+	c, err := ba.bot.Listen(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("[bot] Start listening to subscribe messages...")
```
Comment (dyng) [inline server/bot/bot.go:56]: Can be replaced by `github.com/ethereum/go-ethereum/log`. We use this logging library for prototyping and can switch to others later.

File: server/bot/bot.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	return &BotApplication{
+		bot:    bot,
+		config: config,
+	}
+}
+
+func (ba *BotApplication) Run(ctx context.Context) error {
+	c, err := ba.bot.Listen(ctx)
+	if err != nil {
+		panic(err)
```
Comment (dyng) [inline server/bot/bot.go:53]: Can replace `panic` with `log.Crit("Cannot listen to xxx", "err", err)`.

File: server/bot/bot.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+	done := make(chan struct{})
+	defer close(done)
+
+	go func(c <-chan nostr.Event) {
+		for ev := range c {
+			if strings.Contains(ev.Content, "#subscribe") {
+				fmt.Printf("[bot] Preparing channel for %s\n", ev.PubKey)
+				ba.bot.GetOrCreateSubSK(ctx, ev.PubKey)
+				ba.bot.SendWelcomeMessage(ctx, ba.config.Bot.SK, ev.PubKey)
```
Comment (dyng) [inline server/bot/bot.go:66]: Take care of event duplication. As the event channel (result of `client.Subscribe`) is an aggregated channel of multiple relays, there is always duplicated event of same event sent to multiple relays.

File: server/bot/bot.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+				"p": []string{b.pub},
+			},
+		},
+	}
+	return b.client.Subscribe(ctx, filters), nil
+}
+
+func (b *Bot) GetOrCreateSubSK(ctx context.Context, userPub string) (string, bool, error) {
+	// TODO: use a persistent storage for subSK
+	if subSK, ok := userSubStore[userPub]; ok {
```
Comment (dyng) [inline server/bot/bot.go:106]: Should introduce `postgres` or `sqllite` later.

**Comment by toruta39** [comment]
push off persistence layer to another pull request

**Pull request merged**